routing/localchans/manager.go

```diff
@@ -182,9 +183,28 @@ func (r *Manager) updateEdge(tx kvdb.RTx, chanPoint wire.OutPoint,
 		newSchema.FeeRate,
 	)
 
-	inboundFee := newSchema.InboundFee.ToWire()
-	if err := edge.ExtraOpaqueData.PackRecords(&inboundFee); err != nil {
-		return err
+	// If inbound fees are set, we update the edge with them. Otherwise,
+	// we extract the currently set inbound fee from the ExtraOpaqueData
+	// and use it.
+	if !newSchema.InboundFee.IsNone() {
+		err := fn.MapOptionZ(newSchema.InboundFee,
+			func(f models.InboundFee) error {
+				edge.InboundFee = f
+				inboundWireFee := f.ToWire()
+				return edge.ExtraOpaqueData.PackRecords(
+					&inboundWireFee,
+				)
+			})
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
`newSchema` only contains an `InboundFee` if it is to be updated. Otherwise we have to use and determine the existing InboundFee. Therefore, there must be a case somewhere.